### ISSUE TITLE
Implement remote generation of signing keys for LockKeeperClient

### ISSUE THREAD

**Issue by indomitableSwan** [boltlabs-inc/key-mgmt#187]
**Background:** In the Arbitrary Secrets PoC, we provide the capability to locally generate (and encrypt) a key on the client, followed by sending it to the Key Server for secure storage. Our intention is to add a new capability to the Key Server, such that the key generation (and encryption) occurs remotely on the server. Refer to this PR on the Key Management spec for more details: https://github.com/boltlabs-inc/key-mgmt-spec/pull/111.

**Requirements:**

1. New `LockKeeperClient` API: 
   - Add a new public function to the local client (formerly called the DamsClient) API that allows the caller to perform a remote generation of a signing key
   - If a valid session does not exist, then return an appropriate error code that authentication is required
   - API function name: TBD (example: `generate_signing_key`)
   - Outputs: KeyId, LockKeeperClientError
1. Client Action: Add a new ClientAction type to send a message between LockKeeperClient and Key Server to request a remote key generation.
1. New Server Feature: Implement feature that will process the remote key generation request as follows:
   - generate a signing key pair
       - This secret MUST be randomly generated according to the uniform distribution (refer to spec)
       - This secret should default to 256-bits (refer to spec)
   - encrypt the key pair
   - store in secure storage
   - return key_id and public_key to the caller
1. Provide unit test(s) for the Key Server. For example:
   - generate, encrypt and store key
   - retrieve and decrypt key
   - compare retrieved key to the originally generated key

**Design Thoughts:** Refer to design diagram in Confluence (associated with Remote API Design).

**For QA:** This section is TBD, but will include how we intend to test these changes to determine whether the work is complete.

**Future Thoughts (not part of this ticket):**  We may want to add this capability to the Demo App and/or interactive client. If so, then a new ticket should be created and associated with this issue.

**Comment by marsella** [comment]
NB: the key generation protocol should result in the client getting back the public component of the new key. See boltlabs-inc/key-mgmt-spec#135 - at time of writing, this isn't explicit in the spec.

**Comment by picklenerd** [comment]
Note that there is no such thing as a valid session as of right now. We will be able to implement the most basic version of that requirement after #213 is finished.
